pkg/kotsclient/cluster_create.go

```diff
@@ -73,8 +81,19 @@ func (c *VendorV3Client) CreateCluster(opts CreateClusterOpts) (*types.Cluster,
 	cluster := CreateClusterResponse{}
 	err := c.DoJSON("POST", "/v3/cluster", http.StatusCreated, reqBody, &cluster)
 	if err != nil {
-		return nil, err
+		if strings.Contains(err.Error(), " 400: ") {
+			// to avoid a lot of brittle string parsing, we make the request again with
+			// a dry-run flag and expect to get the same result back
+			ve := ValidationError{}
+			err := c.DoJSON("POST", "/v3/cluster?dry-run=true", http.StatusOK, reqBody, &ve)
+			if err != nil {
+				return nil, nil, err
+			}
+
+			return nil, &ve, nil
```

Comment:
lgtm, but this would be more Go-style error handling if this function returned a custom error type instead of the additional return value

```return nil, CreateClusterErrors{Errors: ve.Errors}```